src/data_types/tensor.hpp

```diff
@@ -0,0 +1,432 @@
+ * @tparam Dims The dimensions that can be used to index the vector.
+ */
+template <class... Dims>
+using DVector = Vector<double, Dims...>;
+
+} // namespace tensor_tools
+
+//////////////////////////////////////////////////////////////////////////
+//                         Operators
+//////////////////////////////////////////////////////////////////////////
+
+namespace ddcHelper {
+
+/**
+ * @brief A helper function to get a modifiable reference to an element of the tensor.
+ * @tparam QueryIndexTag A type describing the relevant index.
+ * @param tensor The tensor whose elements are examined.
+ * @return The relevant element of the tensor.
+ */
+template <class... QueryIndexTag, class ElementType, class... ValidIndexSet>
+KOKKOS_INLINE_FUNCTION ElementType& get(Tensor<ElementType, ValidIndexSet...>& tensor)
+{
+    return tensor.template get<tensor_tools::TensorIndexElement<
+            ddc::detail::TypeSeq<ValidIndexSet...>,
+            QueryIndexTag...>>();
+}
+
+/**
+ * @brief A helper function to get an element of the tensor.
+ * @tparam QueryIndexTag A type describing the relevant index.
+ * @param tensor The tensor whose elements are examined.
+ * @return The relevant element of the tensor.
+ */
+template <class... QueryIndexTag, class ElementType, class... ValidIndexSet>
+KOKKOS_INLINE_FUNCTION ElementType get(Tensor<ElementType, ValidIndexSet...> const& tensor)
+{
+    return tensor.template get<tensor_tools::TensorIndexElement<
+            ddc::detail::TypeSeq<ValidIndexSet...>,
+            QueryIndexTag...>>();
+}
+
+/**
+ * @brief A helper function to convert a vector to a coordinate.
+ * This is useful in order to add a Vector to a coordinate to obtain a
+ * new coordinate (e.g. when calculating the foot of a characteristic.
+ * @param tensor The tensor to be converted.
+ * @return The new coordinate.
+ */
+template <class ElementType, class... Dims>
+KOKKOS_INLINE_FUNCTION Coord<Dims...> to_coord(Vector<ElementType, Dims...> tensor)
+{
+    return Coord<Dims...>(get<Dims>(tensor)...);
+}
+} // namespace ddcHelper
+
+/**
+ * @brief An operator to multiply all the element of the current tensor by
+ * a value.
+ * @param val The value by which the elements should be multiplied.
+ * @return A new tensor containing the result of the multiplication.
+ */
+template <class ElementType, class OElementType, class... ValidIndexSet>
+KOKKOS_INLINE_FUNCTION Tensor<ElementType, ValidIndexSet...> operator*(
+        OElementType val,
+        Tensor<ElementType, ValidIndexSet...> tensor)
+{
+    return tensor * val;
+}
+
+/**
+ * An operator to add the elements of a tensor to a coordinate.
+ * This can be useful in some calculations, e.g when calculating the foot
+ * of a characteristic.
+ * @param tensor The tensor to be converted.
+ * @return The new coordinate.
+ */
+template <class... Dims>
+KOKKOS_INLINE_FUNCTION Coord<Dims...> operator+(Coord<Dims...> coord, DVector<Dims...> tensor)
```

Comment:
```suggestion
KOKKOS_INLINE_FUNCTION Coord<Dims...> operator+(Coord<Dims...> const& coord, DVector<Dims...> const& tensor)
```

Same for other operators.